src/libexpr/attr-path.hh

```diff
@@ -14,7 +14,8 @@ std::pair<Value *, Pos> findAlongAttrPath(
     EvalState & state,
     const std::string & attrPath,
     Bindings & autoArgs,
```

Comment:
These are only needed when `callFunctors` is true, aren’t they?
Maybe we could change the function signature to make `autoArgs` be a `std::optional<Bindings>` and only call he functors if the optional isn’t empty (or even better make this a proper sum type, but c++ makes them a bit painful to write and manipulate)

Reply:
Maybe `Bindings *` instead of `std::optional<Bindings>`, otherwise the bindings are passed by value.